### ISSUE TITLE
Differentiable approaches

### ISSUE THREAD

**Issue by turian** [jorshi/spiegel#10]
Is there any desire to make this synthesizer differentiable, i.e. porting it to pytorch? Then it could be used for a variety of *inverse* synthesis problems, using gradient-descent and deep learning.

**Comment by jorshi** [comment]
@turian - Definitely! This is actually the direction I am planning on taking this in the new year when I have more time for development. I was planning on using the [magenta DDSP](https://github.com/magenta/ddsp) modules with TensorFlow for this. Do you see benefits to using PyTorch over TensorFlow for this?

**Comment by turian** [comment]
I find PyTorch is easier to grok and more concise. Tensorflow is more popular by people who work at Google. But try both if you don't believe me.

Just be careful about how you implement bandpass / lowpass / biquad features. Pytorch can be quite slow on them and even torchaudio doesn't support it correctly: https://github.com/pytorch/audio/issues/704

Inverse synthesis is a topic I am worse on quite heavily, and the difficulty in getting a good difference LFO has been a persistent question. If you want to talk about any of these problems, lastname at gmail dot com

**Comment by turian** [comment]
@jorshi The main downside of PyTorch is that it it does not have full-TPU compatibility. This feature is slated for early 2021, so not too long that I have adopted tf for when I need very high speed.
